plugins/clipboard/plugin.js

```diff
@@ -502,6 +572,29 @@
 					};
 				} );
 			}
+
+			// Detect if any of paste buttons was touched. In such case we assume that user is using
+			// touch device and force displaying paste dialog (#595).
+			if ( editor.ui.addButton ) {
+				// Waiting for editor instance to be ready seems to be the most reliable way to
+				// be sure that paste buttons are already created.
+				editor.once( 'instanceReady', function() {
+					if ( !editor._.pasteButtons ) {
+						return;
+					}
+
+					CKEDITOR.tools.array.forEach( editor._.pasteButtons, function( name ) {
+						var pasteButton = editor.ui.get( name ),
```

Comment:
This function might return `undefined`, for instance you could have clipboard plugin, but no "Paste" (or related) button in the toolbar.

Reply:
In such case `editor._.pasteButtons` should be empty or undefined.

Reply:
Please check [/tests/plugins/clipboard/manual/missingbutton manual test](http://tests.ckeditor.test:1030/tests/plugins/clipboard/manual/missingbutton) in [t/595-nobutton](https://github.com/ckeditor/ckeditor-dev/tree/t/595-nobutton) branch.